// @arnog/colors/dist/index.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  BACKGROUND_COLORS: () => BACKGROUND_COLORS,
  CATEGORICAL_PALETTES: () => CATEGORICAL_PALETTES,
  COLOR_SCALE_PRESETS: () => COLOR_SCALE_PRESETS,
  DIVERGING_PALETTES: () => DIVERGING_PALETTES,
  FOREGROUND_COLORS: () => FOREGROUND_COLORS,
  SEQUENTIAL_PALETTES: () => SEQUENTIAL_PALETTES,
  apca: () => apca,
  asColorNumber: () => asColorNumber,
  asRgb: () => asRgb,
  blue: () => blue,
  brown: () => brown,
  cMaxFor: () => cMaxFor,
  colorToCss: () => colorToCss,
  colorToHex: () => colorToHex,
  contrastingColor: () => contrastingColor,
  cyan: () => cyan,
  darkShades: () => darkShades,
  darken: () => darken,
  darkenHex: () => darkenHex,
  gammaCorrect: () => gammaCorrect,
  getColorScaleMode: () => getColorScaleMode,
  getLightShadeMap: () => getLightShadeMap,
  getOptimalHueRange: () => getOptimalHueRange,
  gray: () => gray,
  green: () => green,
  hslToRgb: () => hslToRgb,
  indigo: () => indigo,
  interpolateOklch: () => interpolateOklch,
  inverseGammaCorrect: () => inverseGammaCorrect,
  isLightColor: () => isLightColor,
  isNamedColor: () => isNamedColor,
  isValidColor: () => isValidColor,
  lerpOklch: () => lerpOklch,
  liftColor: () => liftColor,
  lighten: () => lighten,
  lime: () => lime,
  mixColors: () => mixColors,
  oklabToRgb: () => oklabToRgb,
  oklch: () => oklch,
  oklchFromRGB: () => oklchFromRGB,
  oklchToRgb: () => oklchToRgb,
  orange: () => orange,
  parseColor: () => parseColor,
  parseColorToHex: () => parseColorToHex,
  parseColorToRgb: () => parseColorToRgb,
  parseColorToRgb01: () => parseColorToRgb01,
  parseColorToRgba01: () => parseColorToRgba01,
  parseColorWithAlpha: () => parseColorWithAlpha,
  parseHexColor: () => parseHexColor,
  pink: () => pink,
  purple: () => purple,
  red: () => red,
  rgbToHex: () => rgbToHex,
  rgbToHsl: () => rgbToHsl,
  rgbToOklab: () => rgbToOklab,
  rgbToOklch: () => rgbToOklch,
  shade: () => shade,
  shades: () => shades,
  teal: () => teal,
  toDarkColor: () => toDarkColor,
  yellow: () => yellow
});
module.exports = __toCommonJS(index_exports);

// src/conversion.ts
function gammaCorrect(channel) {
  const abs = Math.abs(channel);
  if (abs <= 31308e-7) return 12.92 * channel;
  const sign = Math.sign(channel) || 1;
  return sign * (1.055 * Math.pow(abs, 1 / 2.4) - 0.055);
}
function inverseGammaCorrect(channel) {
  const abs = Math.abs(channel);
  if (abs <= 0.04045) return channel / 12.92;
  const sign = Math.sign(channel) || 1;
  return sign * Math.pow((abs + 0.055) / 1.055, 2.4);
}
function hslToRgb(h, s, l) {
  h = h / 360;
  let r, g, b;
  if (s === 0) r = g = b = l;
  else {
    const hue2rgb = (p2, q2, t) => {
      if (t < 0) t += 1;
      if (t > 1) t -= 1;
      if (t < 1 / 6) return p2 + (q2 - p2) * 6 * t;
      if (t < 1 / 2) return q2;
      if (t < 2 / 3) return p2 + (q2 - p2) * (2 / 3 - t) * 6;
      return p2;
    };
    const q = l < 0.5 ? l * (1 + s) : l + s - l * s;
    const p = 2 * l - q;
    r = hue2rgb(p, q, h + 1 / 3);
    g = hue2rgb(p, q, h);
    b = hue2rgb(p, q, h - 1 / 3);
  }
  return {
    r: Math.round(r * 255),
    g: Math.round(g * 255),
    b: Math.round(b * 255)
  };
}
function rgbToHsl(r, g, b) {
  r /= 255;
  g /= 255;
  b /= 255;
  const max = Math.max(r, g, b);
  const min = Math.min(r, g, b);
  const l = (max + min) / 2;
  if (max === min) return { h: 0, s: 0, l };
  const d = max - min;
  const s = l > 0.5 ? d / (2 - max - min) : d / (max + min);
  let h;
  if (max === r) h = ((g - b) / d + (g < b ? 6 : 0)) / 6;
  else if (max === g) h = ((b - r) / d + 2) / 6;
  else h = ((r - g) / d + 4) / 6;
  return { h: h * 360, s, l };
}
function rgbToHex(r, g, b) {
  const cl = (v) => Math.max(0, Math.min(255, Math.round(v)));
  return `#${cl(r).toString(16).padStart(2, "0")}${cl(g).toString(16).padStart(2, "0")}${cl(b).toString(16).padStart(2, "0")}`;
}
function parseHexColor(s) {
  const hex = s.startsWith("#") ? s.substring(1) : s;
  let r, g, b;
  let alpha;
  if (hex.length === 3) {
    r = parseInt(hex[0] + hex[0], 16);
    g = parseInt(hex[1] + hex[1], 16);
    b = parseInt(hex[2] + hex[2], 16);
  } else if (hex.length === 6) {
    r = parseInt(hex.substring(0, 2), 16);
    g = parseInt(hex.substring(2, 4), 16);
    b = parseInt(hex.substring(4, 6), 16);
  } else if (hex.length === 8) {
    r = parseInt(hex.substring(0, 2), 16);
    g = parseInt(hex.substring(2, 4), 16);
    b = parseInt(hex.substring(4, 6), 16);
    alpha = parseInt(hex.substring(6, 8), 16) / 255;
  } else {
    return { r: 0, g: 0, b: 0 };
  }
  const result = { r, g, b };
  if (alpha !== void 0) result.alpha = alpha;
  return result;
}
function asRgb(color) {
  if (typeof color === "number") {
    return {
      r: color >>> 24 & 255,
      g: color >>> 16 & 255,
      b: color >>> 8 & 255,
      alpha: (color & 255) / 255
    };
  }
  if (typeof color === "string") return parseHexColor(color);
  if ("C" in color) return oklchToRgb(color);
  if ("a" in color && "b" in color) return oklabToRgb(color);
  return color;
}
function asColorNumber(color) {
  const rgb = asRgb(color);
  const a = rgb.alpha !== void 0 ? clampByte(rgb.alpha * 255) : 255;
  return (clampByte(rgb.r) << 24 | clampByte(rgb.g) << 16 | clampByte(rgb.b) << 8 | a) >>> 0;
}
function clampByte(v) {
  if (v < 0) return 0;
  if (v > 255) return 255;
  return Math.round(v);
}
function colorToHex(color) {
  const toHex = (n) => {
    const hex = Math.round(Math.max(0, Math.min(255, n))).toString(16);
    return hex.length === 1 ? "0" + hex : hex;
  };
  const r = color >>> 24 & 255;
  const g = color >>> 16 & 255;
  const b = color >>> 8 & 255;
  const a = color & 255;
  if (a !== 255) return `#${toHex(r)}${toHex(g)}${toHex(b)}${toHex(a)}`;
  return `#${toHex(r)}${toHex(g)}${toHex(b)}`;
}
function oklchToOklab(_) {
  const [L, C, H] = [_.L, _.C, _.H];
  const hRadians = H * Math.PI / 180;
  const result = {
    L,
    a: C * Math.cos(hRadians),
    b: C * Math.sin(hRadians)
  };
  if (_.alpha !== void 0) result.alpha = _.alpha;
  return result;
}
function oklabToOklch(_) {
  const [L, a, b] = [_.L, _.a, _.b];
  const C = Math.sqrt(a * a + b * b);
  const hRadians = Math.atan2(b, a);
  const H = hRadians * 180 / Math.PI;
  const result = { L, C, H };
  if (_.alpha !== void 0) result.alpha = _.alpha;
  return result;
}
function oklabToUnclippedRgb(_) {
  const [l, a, b] = [_.L, _.a, _.b];
  const L = Math.pow(
    0.9999999984505198 * l + 0.39633779217376786 * a + 0.2158037580607588 * b,
    3
  );
  const M = Math.pow(
    1.00000000888176 * l - 0.10556134232365635 * a - 0.0638541747717059 * b,
    3
  );
  const S = Math.pow(
    l * 1.000000054672411 - 0.0894841820949657 * a - 1.2914855378640917 * b,
    3
  );
  const r = 4.076741661347994 * L - 3.307711590408193 * M + 0.230969928729428 * S;
  const g = -1.2684380040921763 * L + 2.6097574006633715 * M - 0.3413193963102197 * S;
  const bl = -0.004196086541837188 * L - 0.7034186144594493 * M + 1.7076147009309444 * S;
  return [gammaCorrect(r), gammaCorrect(g), gammaCorrect(bl)];
}
function inGamut(rgb) {
  const [r, g, b] = rgb;
  return r >= 0 && r <= 1 && g >= 0 && g <= 1 && b >= 0 && b <= 1;
}
function clampRgb(rgb, alpha) {
  let [r, g, b] = rgb;
  r = clampByte(r * 255);
  g = clampByte(g * 255);
  b = clampByte(b * 255);
  return alpha !== void 0 ? { r, g, b, alpha } : { r, g, b };
}
function oklabToRgb(color) {
  let [r, g, b] = oklabToUnclippedRgb(color);
  if (inGamut([r, g, b])) return clampRgb([r, g, b], color.alpha);
  const oklch2 = oklabToOklch(color);
  oklch2.C = 0;
  [r, g, b] = oklabToUnclippedRgb(oklchToOklab(oklch2));
  if (!inGamut([r, g, b])) return clampRgb([r, g, b], color.alpha);
  let low = 0;
  let high = color.L;
  let mid = (low + high) / 2;
  oklch2.C = mid;
  const resolution = 0.36 / Math.pow(2, 12);
  while (high - low > resolution) {
    mid = (low + high) / 2;
    oklch2.C = mid;
    [r, g, b] = oklabToUnclippedRgb(oklchToOklab(oklch2));
    if (inGamut([r, g, b])) low = mid;
    else high = mid;
  }
  return clampRgb([r, g, b], color.alpha);
}
function oklchToRgb(_) {
  return oklabToRgb(oklchToOklab(_));
}
function rgbToOklab(_) {
  const [r, g, b] = [_.r, _.g, _.b];
  const rLin = inverseGammaCorrect(r / 255);
  const gLin = inverseGammaCorrect(g / 255);
  const bLin = inverseGammaCorrect(b / 255);
  const L = 0.41222147079999993 * rLin + 0.5363325363 * gLin + 0.0514459929 * bLin;
  const M = 0.2119034981999999 * rLin + 0.6806995450999999 * gLin + 0.1073969566 * bLin;
  const S = 0.08830246189999998 * rLin + 0.2817188376 * gLin + 0.6299787005000002 * bLin;
  const L3 = Math.cbrt(L);
  const M3 = Math.cbrt(M);
  const S3 = Math.cbrt(S);
  const result = {
    L: 0.2104542553 * L3 + 0.793617785 * M3 - 0.0040720468 * S3,
    a: 1.9779984951 * L3 - 2.428592205 * M3 + 0.4505937099 * S3,
    b: 0.0259040371 * L3 + 0.7827717662 * M3 - 0.808675766 * S3
  };
  if (_.alpha !== void 0) result.alpha = _.alpha;
  return result;
}
function rgbToOklch(_) {
  return oklabToOklch(rgbToOklab(_));
}
function cMaxFor(L, Hdeg) {
  const hRad = Hdeg * Math.PI / 180;
  let lo = 0;
  let hi = 1.5;
  for (let i = 0; i < 32; i++) {
    const C = (lo + hi) / 2;
    const [r, g, b] = oklabToUnclippedRgb({
      L,
      a: C * Math.cos(hRad),
      b: C * Math.sin(hRad)
    });
    if (r >= 0 && r <= 1 && g >= 0 && g <= 1 && b >= 0 && b <= 1) lo = C;
    else hi = C;
  }
  return lo;
}
function oklch(l, c, h) {
  const L = l / 100;
  if (L <= 0) return 255;
  const rgb = oklchToRgb({ L, C: c, H: h });
  return (clampByte(rgb.r) << 24 | clampByte(rgb.g) << 16 | clampByte(rgb.b) << 8 | 255) >>> 0;
}
function toDarkColor(color) {
  const { l, c, h } = oklchFromRGB(color);
  return oklch(l + 2.8, c + 2e-3, h + 0.46);
}
function oklchFromRGB(rgba) {
  const r = rgba >>> 24 & 255;
  const g = rgba >>> 16 & 255;
  const b = rgba >>> 8 & 255;
  const result = rgbToOklch({ r, g, b });
  return {
    l: result.L * 100,
    c: result.C,
    h: result.H < 0 ? result.H + 360 : result.H
  };
}

// src/interpolation.ts
function interpolateColor(color1, color2, t) {
  const { l: l1, c: c1, h: h1 } = oklchFromRGB(color1);
  const { l: l2, c: c2, h: h2 } = oklchFromRGB(color2);
  const l = l1 + (l2 - l1) * t;
  const c = c1 + (c2 - c1) * t;
  let hDiff = h2 - h1;
  if (hDiff > 180) hDiff -= 360;
  if (hDiff < -180) hDiff += 360;
  const h = h1 + hDiff * t;
  const a1 = color1 & 255;
  const a2 = color2 & 255;
  const a = Math.round(a1 + (a2 - a1) * t);
  const newColor = oklch(l, c, h < 0 ? h + 360 : h > 360 ? h - 360 : h);
  return newColor & 4294967040 | a;
}
function getOptimalHueRange(hue, chromaBackoff = 0.1) {
  const targetDarkChroma = 0.15;
  const targetLightChroma = 0.02;
  const darkLightnessMin = 0.35;
  const darkLightnessMax = 0.45;
  const lightLightnessMax = 0.96;
  let bestDarkL = darkLightnessMax;
  let bestDarkChroma = 0;
  for (let L = darkLightnessMax; L >= darkLightnessMin; L -= 0.01) {
    const maxC = cMaxFor(L, hue);
    if (maxC >= targetDarkChroma) {
      bestDarkL = L;
      bestDarkChroma = targetDarkChroma;
      break;
    } else if (maxC > bestDarkChroma) {
      bestDarkL = L;
      bestDarkChroma = maxC;
    }
  }
  if (bestDarkChroma < targetDarkChroma * 0.8) {
    for (let L = darkLightnessMax + 0.01; L <= 0.5; L += 0.01) {
      const maxC = cMaxFor(L, hue);
      if (maxC >= targetDarkChroma || maxC > bestDarkChroma * 1.2) {
        bestDarkL = L;
        bestDarkChroma = Math.min(targetDarkChroma, maxC);
        break;
      }
    }
  }
  const bestLightL = lightLightnessMax;
  let bestLightChroma = targetLightChroma;
  const maxLightChroma = cMaxFor(lightLightnessMax, hue);
  if (maxLightChroma < targetLightChroma) {
    bestLightChroma = maxLightChroma * (1 - chromaBackoff);
  }
  bestLightChroma = Math.max(
    0.015,
    Math.min(targetLightChroma, bestLightChroma)
  );
  const darkest = oklch(bestDarkL * 100, bestDarkChroma, hue);
  const lightest = oklch(bestLightL * 100, bestLightChroma, hue);
  return { darkest, lightest };
}
function shade(color, index) {
  const idx = Math.max(0, Math.min(1e3, index));
  let darkest, mid, lightest;
  if (typeof color === "object") {
    darkest = color.darkest;
    lightest = color.lightest;
    mid = color.mid ?? 0;
  } else {
    mid = color;
    const { h } = oklchFromRGB(mid);
    const alpha = mid & 255;
    const { darkest: optimalDarkest, lightest: optimalLightest } = getOptimalHueRange(h);
    darkest = optimalDarkest & 4294967040 | alpha;
    lightest = optimalLightest & 4294967040 | alpha;
  }
  if (!mid && typeof color === "object" && !color.mid)
    mid = interpolateColor(darkest, lightest, 0.5);
  if (idx <= 500) {
    const localT2 = idx / 500;
    const smoothT2 = localT2 * localT2 * (3 - 2 * localT2);
    return interpolateColor(lightest, mid, smoothT2);
  }
  const localT = (idx - 500) / 500;
  const smoothT = localT * localT * (3 - 2 * localT);
  return interpolateColor(mid, darkest, smoothT);
}

// src/scale-presets.ts
var COLOR_SCALE_PRESETS = [
  {
    id: "pink",
    name: "Pink",
    colors: {
      mid: { lightness: 69, chroma: 0.202, hue: 7.4 }
    }
  },
  {
    id: "red",
    name: "Red",
    colors: {
      lightest: { lightness: 95, chroma: 0.2, hue: 30 },
      mid: { lightness: 62, chroma: 0.25, hue: 30 },
      darkest: { lightness: 36, chroma: 0.16, hue: 25 }
    }
  },
  {
    id: "brown",
    name: "Brown",
    colors: {
      lightest: { lightness: 94.5, chroma: 0.021, hue: 72.1 },
      mid: { lightness: 59, chroma: 0.2, hue: 73 },
      darkest: { lightness: 20, chroma: 0.061, hue: 72.7 }
    }
  },
  // {
  //   id: "orange2",
  //   name: "Orange 2",
  //   theme: "light",
  //   colors: {
  //     lightest: { lightness: 95, chroma: 0.1, hue: 73 },
  //     mid: { lightness: 67, chroma: 0.23, hue: 54 },
  //     darkest: { lightness: 43, chroma: 0.2, hue: 37 },
  //   },
  // },
  {
    id: "orange",
    name: "Orange",
    colors: {
      lightest: { lightness: 95, chroma: 0.06, hue: 82 },
      mid: { lightness: 73, chroma: 0.21, hue: 56 },
      darkest: { lightness: 46, chroma: 0.2, hue: 35 }
    }
  },
  {
    id: "yellow",
    name: "Yellow",
    // colors: {
    //   lightest: { lightness: 95, chroma: 0.15, hue: 106 },
    //   mid: { lightness: 67, chroma: 0.28, hue: 87 },
    //   darkest: { lightness: 48, chroma: 0.11, hue: 52 },
    // },
    colors: {
      lightest: { lightness: 97, chroma: 0.02, hue: 106 },
      mid: { lightness: 89, chroma: 0.21, hue: 99 },
      darkest: { lightness: 67, chroma: 0.43, hue: 59 }
    }
  },
  {
    id: "lime",
    name: "Lime",
    colors: {
      mid: { lightness: 65, chroma: 0.192, hue: 134.3 }
    }
  },
  {
    id: "green",
    name: "Green",
    colors: {
      mid: { lightness: 64, chroma: 0.21, hue: 144 }
    }
  },
  {
    id: "teal",
    name: "Teal",
    colors: {
      mid: { lightness: 65, chroma: 0.116, hue: 192.6 }
    }
  },
  {
    id: "cyan",
    name: "Cyan",
    colors: {
      lightest: { lightness: 97, chroma: 0.11, hue: 195 },
      mid: { lightness: 61, chroma: 0.11, hue: 210 },
      darkest: { lightness: 42, chroma: 0.11, hue: 210 }
    }
  },
  {
    id: "blue",
    name: "Blue",
    colors: {
      lightest: { lightness: 91, chroma: 0.22, hue: 240 },
      mid: { lightness: 63, chroma: 0.22, hue: 255 },
      darkest: { lightness: 38, chroma: 0.22, hue: 252 }
    }
  },
  {
    id: "indigo",
    name: "Indigo",
    colors: {
      lightest: { lightness: 92.7, chroma: 0.036, hue: 291 },
      mid: { lightness: 67, chroma: 0.285, hue: 296 },
      darkest: { lightness: 28, chroma: 0.23, hue: 278 }
    }
  },
  {
    id: "purple",
    name: "Purple",
    colors: {
      mid: { lightness: 67, chroma: 0.21, hue: 299 }
    }
  }
];
function getColorScaleMode(preset) {
  const { colors } = preset;
  const hasLight = Boolean(colors.lightest);
  const hasDark = Boolean(colors.darkest);
  if (hasLight && hasDark && colors.mid) {
    return "three";
  }
  if (hasLight || hasDark) {
    return "two";
  }
  return "single";
}

// src/scale.ts
var SHADES = [50, 100, 200, 300, 400, 500, 600, 700, 800, 900];
function stopToNumber({ lightness, chroma, hue }) {
  return oklch(lightness, chroma, hue);
}
function presetToShadeInput(preset) {
  const { mid, lightest, darkest } = preset.colors;
  if (lightest && darkest && mid) {
    return {
      lightest: stopToNumber(lightest),
      mid: stopToNumber(mid),
      darkest: stopToNumber(darkest)
    };
  }
  if (lightest && darkest) {
    return {
      lightest: stopToNumber(lightest),
      darkest: stopToNumber(darkest)
    };
  }
  if (mid) return stopToNumber(mid);
  throw new Error(
    `Color scale preset "${preset.id}" is missing required anchors.`
  );
}
var _lightShadeMap = null;
function getLightShadeMap() {
  if (!_lightShadeMap) {
    _lightShadeMap = /* @__PURE__ */ Object.create(null);
    for (const preset of COLOR_SCALE_PRESETS) {
      _lightShadeMap[preset.id] = presetToShadeInput(preset);
    }
  }
  return _lightShadeMap;
}
function paletteColor(id, n = 500) {
  return colorToHex(shade(getLightShadeMap()[id], n));
}
function shades(fn) {
  return Object.fromEntries(SHADES.map((n) => [n, fn(n)]));
}
function darkShades(fn) {
  return Object.fromEntries(
    SHADES.map((n) => [n, colorToHex(toDarkColor(asColorNumber(fn(n))))])
  );
}
function yellow(n = 500) {
  return paletteColor("yellow", n);
}
function brown(n = 500) {
  return paletteColor("brown", n);
}
function red(n = 500) {
  return paletteColor("red", n);
}
function orange(n = 500) {
  return paletteColor("orange", n);
}
function lime(n = 500) {
  return paletteColor("lime", n);
}
function green(n = 500) {
  return paletteColor("green", n);
}
function pink(n = 500) {
  return paletteColor("pink", n);
}
function purple(n = 500) {
  return paletteColor("purple", n);
}
function indigo(n = 500) {
  return paletteColor("indigo", n);
}
function blue(n = 500) {
  return paletteColor("blue", n);
}
function teal(n = 500) {
  return paletteColor("teal", n);
}
function cyan(n = 500) {
  return paletteColor("cyan", n);
}

// src/parsing.ts
var NAMED_COLORS = {
  red: "#d7170b",
  //<- 700, 500 ->'#f21c0d'
  orange: "#fe8a2b",
  yellow: "#ffc02b",
  // <- 600, 500 -> '#ffcf33',
  lime: "#63b215",
  green: "#21ba3a",
  teal: "#17cfcf",
  cyan: "#13a7ec",
  blue: "#0d80f2",
  indigo: "#63c",
  purple: "#a219e6",
  magenta: "#eb4799",
  brown: "#8c564b",
  olive: "#8a8f2a",
  midnight: "#2c4670",
  sky: "#d2dce9",
  black: "#000",
  white: "#ffffff",
  carbon: "#111111",
  // near-black, high-contrast text
  charcoal: "#333333",
  // primary axis / label color
  slate: "#555555",
  // secondary text, major gridlines
  "dark-grey": "#666",
  graphite: "#777777",
  // minor gridlines
  stone: "#999999",
  // de-emphasized strokes
  grey: "#A6A6A6",
  "light-grey": "#d4d5d2",
  ash: "#E6E6E6",
  // subtle fills, light strokes
  mist: "#F3F3F3",
  // light background tint
  snow: "#FFFFFF"
  // pure white
};
function isNamedColor(s) {
  const str = s.trim().toLowerCase();
  return str === "transparent" || str in NAMED_COLORS;
}
function parseColor(s, darkMode) {
  const str = s.trim().toLowerCase();
  if (str.startsWith("#")) {
    const hex = str.substring(1);
    let r, g, b, a = 255;
    if (hex.length === 3) {
      r = parseInt(hex[0] + hex[0], 16);
      g = parseInt(hex[1] + hex[1], 16);
      b = parseInt(hex[2] + hex[2], 16);
    } else if (hex.length === 6) {
      r = parseInt(hex.substring(0, 2), 16);
      g = parseInt(hex.substring(2, 4), 16);
      b = parseInt(hex.substring(4, 6), 16);
    } else if (hex.length === 8) {
      r = parseInt(hex.substring(0, 2), 16);
      g = parseInt(hex.substring(2, 4), 16);
      b = parseInt(hex.substring(4, 6), 16);
      a = parseInt(hex.substring(6, 8), 16);
    } else return 0;
    return (r << 24 | g << 16 | b << 8 | a) >>> 0;
  }
  const rgbMatch = str.match(/^rgba?\s*\(\s*([^)]+)\s*\)$/);
  if (rgbMatch) {
    const parts = rgbMatch[1].replace(/[,/]/g, " ").trim().split(/\s+/);
    let r = parseFloat(parts[0]) || 0;
    let g = parseFloat(parts[1]) || 0;
    let b = parseFloat(parts[2]) || 0;
    if (parts[0].includes("%") || parts[1].includes("%") || parts[2].includes("%")) {
      if (parts[0].includes("%")) r = r / 100 * 255;
      if (parts[1].includes("%")) g = g / 100 * 255;
      if (parts[2].includes("%")) b = b / 100 * 255;
    } else if (r <= 1 && g <= 1 && b <= 1) {
      r = r * 255;
      g = g * 255;
      b = b * 255;
    }
    r = Math.round(Math.max(0, Math.min(255, r)));
    g = Math.round(Math.max(0, Math.min(255, g)));
    b = Math.round(Math.max(0, Math.min(255, b)));
    let a = 255;
    if (parts.length >= 4) {
      let alpha = parseFloat(parts[3]);
      if (parts[3].includes("%")) alpha = alpha / 100;
      else if (alpha > 1) alpha = alpha / 255;
      a = Math.round(Math.max(0, Math.min(255, alpha * 255)));
    }
    return (r << 24 | g << 16 | b << 8 | a) >>> 0;
  }
  const oklchMatch = str.match(/^oklch\s*\(\s*([^)]+)\s*\)$/);
  if (oklchMatch) {
    const parts = oklchMatch[1].replace(/[,/]/g, " ").trim().split(/\s+/);
    let l = parseFloat(parts[0]);
    if (parts[0].includes("%")) l = l / 100;
    else if (l <= 1) {
    } else {
      l = l / 100;
    }
    const c = parseFloat(parts[1]) || 0;
    const h = parseFloat(parts[2]) || 0;
    let alpha = 1;
    if (parts.length >= 4) {
      alpha = parseFloat(parts[3]);
      if (parts[3].includes("%")) alpha = alpha / 100;
      else if (alpha > 1) {
        alpha = alpha / 255;
      }
    }
    const rgb = oklch(l * 100, c, h);
    const a = Math.round(alpha * 255);
    return rgb & 4294967040 | a;
  }
  const oklabMatch = str.match(/^oklab\s*\(\s*([^)]+)\s*\)$/);
  if (oklabMatch) {
    const parts = oklabMatch[1].replace(/[,/]/g, " ").trim().split(/\s+/);
    let l = parseFloat(parts[0]);
    if (parts[0].includes("%")) l = l / 100;
    else if (l > 1) l = l / 100;
    const labA = parseFloat(parts[1]) || 0;
    const labB = parseFloat(parts[2]) || 0;
    let alpha = 1;
    if (parts.length >= 4) {
      alpha = parseFloat(parts[3]);
      if (parts[3].includes("%")) alpha = alpha / 100;
      else if (alpha > 1) alpha = alpha / 255;
    }
    const rgb = oklabToRgb({ L: l, a: labA, b: labB });
    const rByte = rgb.r;
    const gByte = rgb.g;
    const bByte = rgb.b;
    const alphaByte = Math.round(alpha * 255);
    return (rByte << 24 | gByte << 16 | bByte << 8 | alphaByte) >>> 0;
  }
  const hslMatch = str.match(/^hsl\s*\(\s*([^)]+)\s*\)$/);
  if (hslMatch) {
    const parts = hslMatch[1].replace(/[,/]/g, " ").trim().split(/\s+/);
    const h = parseFloat(parts[0]) || 0;
    let s2 = parseFloat(parts[1]) || 0;
    if (parts[1].includes("%")) s2 = s2 / 100;
    else if (s2 <= 1) {
    } else {
      s2 = s2 / 100;
    }
    let l = parseFloat(parts[2]) || 0;
    if (parts[2].includes("%")) l = l / 100;
    else if (l <= 1) {
    } else {
      l = l / 100;
    }
    let alpha = 1;
    if (parts.length >= 4) {
      alpha = parseFloat(parts[3]);
      if (parts[3].includes("%")) alpha = alpha / 100;
      else if (alpha > 1) {
        alpha = alpha / 255;
      }
    }
    const { r, g, b } = hslToRgb(h, s2, l);
    const a = Math.round(alpha * 255);
    return (r << 24 | g << 16 | b << 8 | a) >>> 0;
  }
  if (str === "transparent") return 0;
  const paletteMatch = str.match(/^([a-z]+)-(\d+)$/);
  if (paletteMatch) {
    const input = getLightShadeMap()[paletteMatch[1]];
    if (input) {
      const color = shade(input, parseInt(paletteMatch[2], 10));
      return darkMode ? toDarkColor(color) : color;
    }
  }
  if (str in NAMED_COLORS)
    return parseColor(NAMED_COLORS[str]);
  console.warn(`parseColor: unrecognized color "${s}"`);
  return 0;
}
function parseColorToRgb(s, darkMode) {
  const color = parseColor(s, darkMode);
  return {
    r: color >>> 24 & 255,
    g: color >>> 16 & 255,
    b: color >>> 8 & 255,
    alpha: (color & 255) / 255
  };
}
function parseColorToHex(s, darkMode) {
  const color = parseColor(s, darkMode);
  const clamp = (v) => Math.max(0, Math.min(255, Math.round(v)));
  const rHex = clamp(color >>> 24 & 255).toString(16).padStart(2, "0");
  const gHex = clamp(color >>> 16 & 255).toString(16).padStart(2, "0");
  const bHex = clamp(color >>> 8 & 255).toString(16).padStart(2, "0");
  const alpha = clamp(color & 255).toString(16).padStart(2, "0");
  if (alpha === "ff") return `#${rHex}${gHex}${bHex}`;
  return `#${rHex}${gHex}${bHex}${alpha}`;
}
function parseColorToRgb01(s, darkMode) {
  const color = parseColor(s, darkMode);
  return [
    (color >>> 24 & 255) / 255,
    (color >>> 16 & 255) / 255,
    (color >>> 8 & 255) / 255
  ];
}
function parseColorToRgba01(s, darkMode) {
  const color = parseColor(s, darkMode);
  return [
    (color >>> 24 & 255) / 255,
    (color >>> 16 & 255) / 255,
    (color >>> 8 & 255) / 255,
    (color & 255) / 255
  ];
}
function parseColorWithAlpha(color, alpha, darkMode) {
  const parsed = parseColor(color, darkMode);
  const r = parsed >>> 24 & 255;
  const g = parsed >>> 16 & 255;
  const b = parsed >>> 8 & 255;
  const a = Math.round(Math.max(0, Math.min(1, alpha)) * 255);
  if (a === 255) return rgbToHex(r, g, b);
  return `#${r.toString(16).padStart(2, "0")}${g.toString(16).padStart(2, "0")}${b.toString(16).padStart(2, "0")}${a.toString(16).padStart(2, "0")}`;
}
function isValidColor(s) {
  const str = s.trim().toLowerCase();
  if (str === "transparent") return true;
  if (str in NAMED_COLORS) return true;
  if (str.startsWith("#")) {
    const hex = str.substring(1);
    return [3, 6, 8].includes(hex.length) && /^[0-9a-f]+$/.test(hex);
  }
  if (/^(rgba?|oklch|hsl)\s*\(/.test(str)) return true;
  const palMatch = str.match(/^([a-z]+)-(\d+)$/);
  if (palMatch && palMatch[1] in getLightShadeMap()) return true;
  if (/^[a-z]+$/.test(str)) return true;
  return false;
}

// src/contrast.ts
function apca(bgColor, fgColor) {
  const bgRgb = asRgb(bgColor);
  const fgRgb = asRgb(fgColor);
  const normBG = 0.56;
  const normTXT = 0.57;
  const revTXT = 0.62;
  const revBG = 0.65;
  const blkThrs = 0.022;
  const blkClmp = 1.414;
  const loClip = 0.1;
  const deltaYmin = 5e-4;
  const scaleBoW = 1.14;
  const loBoWoffset = 0.027;
  const scaleWoB = 1.14;
  const loWoBoffset = 0.027;
  function fclamp(Y) {
    return Y >= blkThrs ? Y : Y + (blkThrs - Y) ** blkClmp;
  }
  function linearize(val) {
    const sign = val < 0 ? -1 : 1;
    return sign * Math.pow(Math.abs(val), 2.4);
  }
  const Yfg = fclamp(
    linearize(fgRgb.r / 255) * 0.2126729 + linearize(fgRgb.g / 255) * 0.7151522 + linearize(fgRgb.b / 255) * 0.072175
  );
  const Ybg = fclamp(
    linearize(bgRgb.r / 255) * 0.2126729 + linearize(bgRgb.g / 255) * 0.7151522 + linearize(bgRgb.b / 255) * 0.072175
  );
  let S, C;
  if (Math.abs(Ybg - Yfg) < deltaYmin) C = 0;
  else {
    if (Ybg > Yfg) {
      S = Ybg ** normBG - Yfg ** normTXT;
      C = S * scaleBoW;
    } else {
      S = Ybg ** revBG - Yfg ** revTXT;
      C = S * scaleWoB;
    }
  }
  if (Math.abs(C) < loClip) return 0;
  if (C > 0) return C - loWoBoffset;
  return C + loBoWoffset;
}
function contrastingColor(arg) {
  let bg, fg1, fg2;
  if (typeof arg !== "object" || !("bg" in arg)) {
    bg = asRgb(arg);
    fg1 = "#ffffff";
    fg2 = "#000000";
  } else {
    bg = arg.bg;
    fg1 = arg.fg1;
    fg2 = arg.fg2;
  }
  const contrast1 = Math.abs(apca(fg1, bg));
  const contrast2 = Math.abs(apca(fg2, bg));
  return contrast1 >= contrast2 ? asColorNumber(fg1) : asColorNumber(fg2);
}

// src/utilities.ts
function gray(percent) {
  return oklch(percent, 0, 0);
}
function lighten(color, amount) {
  const { l, c, h } = oklchFromRGB(color);
  const alpha = color & 255;
  const newL = Math.max(0, Math.min(100, l + amount));
  const newColor = oklch(newL, c, h);
  return newColor & 4294967040 | alpha;
}
function darken(color, amount) {
  return lighten(color, -amount);
}
function darkenHex(color, factor) {
  const packed = parseColor(color);
  const { l, c, h } = oklchFromRGB(packed);
  const newColor = oklch(Math.max(0, l * factor), c, h);
  const r = newColor >>> 24 & 255;
  const g = newColor >>> 16 & 255;
  const b = newColor >>> 8 & 255;
  return rgbToHex(r, g, b);
}
function isLightColor(color) {
  return Math.abs(apca(color, "#000000")) > Math.abs(apca(color, "#ffffff"));
}
function mixColors(a, b, t) {
  const f = Math.max(0, Math.min(1, t));
  return [
    a[0] * (1 - f) + b[0] * f,
    a[1] * (1 - f) + b[1] * f,
    a[2] * (1 - f) + b[2] * f
  ];
}
function liftColor(color, lift) {
  if (lift === 0) return color;
  const target = lift > 0 ? [1, 1, 1] : [0, 0, 0];
  return mixColors(color, target, Math.abs(lift));
}
function lerpOklch(c1, c2, f) {
  const L = c1.L + (c2.L - c1.L) * f;
  const C = c1.C + (c2.C - c1.C) * f;
  let dh = c2.H - c1.H;
  if (dh > 180) dh -= 360;
  if (dh < -180) dh += 360;
  let H = c1.H + dh * f;
  if (H < 0) H += 360;
  if (H >= 360) H -= 360;
  return { L, C, H };
}
function interpolateOklch(hex1, hex2, f) {
  const c1 = rgbToOklch(parseHexColor(hex1));
  const c2 = rgbToOklch(parseHexColor(hex2));
  return oklchToRgb(lerpOklch(c1, c2, f));
}
function colorToCss(color) {
  if (color.length === 9) {
    const r = parseInt(color.slice(1, 3), 16);
    const g = parseInt(color.slice(3, 5), 16);
    const b = parseInt(color.slice(5, 7), 16);
    const a = parseInt(color.slice(7, 9), 16) / 255;
    return `rgba(${r}, ${g}, ${b}, ${a})`;
  }
  return color;
}

// src/palettes/categorical.ts
var TYCHO_11 = [
  "#4e79a7",
  // Blue
  "#f28e2b",
  // Orange
  "#59a14f",
  // Green
  "#e15759",
  // Red
  "#b07aa1",
  // Purple
  "#9c755f",
  // Brown
  "#ff9da7",
  // Pink
  "#edc948",
  // Yellow
  "#76b7b2",
  // Teal
  "#6b8fd6",
  // Steel Blue
  "#c07bc4"
  // Magenta
];
var TYCHO_DARK_11 = [
  "#78a6d9",
  // Blue
  "#ffae54",
  // Orange
  "#7ddc7a",
  // Green
  "#ff7a7a",
  // Red
  "#d29be0",
  // Purple
  "#c49a84",
  // Brown
  "#ffb3bf",
  // Pink
  "#ffe066",
  // Yellow
  "#7fd6d0",
  // Teal
  "#8fb4ff",
  // Steel Blue
  "#e199eb"
  // Magenta
];
var TYCHO_ROBUST_11 = [
  "#4e79a7",
  // Blue
  "#f28e2b",
  // Orange
  "#2ca58d",
  // Teal-shifted green (more CB-safe)
  "#d13a3c",
  // Stronger red
  "#b07aa1",
  // Purple
  "#9c755f",
  // Brown
  "#ff9da7",
  // Pink
  "#e3c13b",
  // Darker yellow for contrast
  "#5fb8b2",
  // Teal
  "#6b8fd6",
  // Steel Blue
  "#c07bc4"
  // Magenta
];
var TYCHO_SOFT_11 = [
  "#8fb1d4",
  "#f6b878",
  "#8ecf86",
  "#f08a8b",
  "#d3a9cc",
  "#c3a492",
  "#ffc6cc",
  "#f3e08a",
  "#a8d8d4",
  "#a9c0ea",
  "#e0b4e4"
];
var TYCHO_SOFT_DARK_11 = [
  "#78a6d9",
  "#ffae54",
  "#7ddc7a",
  "#ff7a7a",
  "#d29be0",
  "#c49a84",
  "#ffb3bf",
  "#ffe066",
  "#7fd6d0",
  "#8fb4ff",
  "#e199eb"
];
var TYCHO_BOLD_11 = [
  "#2f6fb0",
  "#ff7a00",
  "#2fa23a",
  "#e02f2f",
  "#9b4db5",
  "#7f4f38",
  "#ff6f86",
  "#f2c200",
  "#2daaa3",
  "#4c79e0",
  "#b84ac6"
];
var TYCHO_BOLD_DARK_11 = [
  "#4f93ff",
  "#ff8c1a",
  "#33c94a",
  "#ff4f4f",
  "#b86bff",
  "#a86a4a",
  "#ff7f9e",
  "#ffd400",
  "#2ec9c1",
  "#6f9bff",
  "#cc5bd9"
];
var MATHEMATICA_10 = [
  "#5E81B5",
  "#E19C24",
  "#8FB131",
  "#EB6235",
  "#8778B3",
  "#C56E1A",
  "#5E9EC9",
  "#B23A3A",
  "#4C9F70",
  "#C979B7"
];
var TABLEAU_10 = [
  "#1f77b4",
  // Blue
  "#ff7f0e",
  // Orange
  "#2ca02c",
  // Green
  "#d62728",
  // Red
  "#9467bd",
  // Purple
  "#8c564b",
  // Brown
  "#e377c2",
  // Pink
  "#7f7f7f",
  // Gray
  "#bcbd22",
  // Olive
  "#17becf"
  // Cyan
];
var CUPERTINO_10 = [
  "#007AFF",
  "#FF9500",
  "#34C759",
  "#FF3B30",
  "#AF52DE",
  "#FF2D55",
  "#30B0C7",
  "#5856D6",
  "#A2845E",
  "#32ADE6",
  "#00C7BE"
];
var CUPERTINO_DARK_10 = [
  "#0A84FF",
  "#FF9F0A",
  "#30D158",
  "#FF453A",
  "#BF5AF2",
  "#FF375F",
  "#40C8E0",
  "#5E5CE6",
  "#AC8E68",
  "#64D2FF",
  "#00D1C1"
];
var KELLY_22 = [
  "#fdfdfd",
  // White
  "#1d1d1d",
  // Black
  "#ebce2b",
  // Yellow
  "#702c8c",
  // Purple
  "#db6917",
  // Orange
  "#96cde6",
  // Light blue
  "#ba1c30",
  // Red
  "#c0bd7f",
  // Buff
  "#7f7e80",
  // Gray
  "#5fa641",
  // Green
  "#d485b2",
  // Purplish pink
  "#4277b6",
  // Blue
  "#df8461",
  // Yellowish pink
  "#463397",
  // Violet
  "#e1a11a",
  // Orange yellow
  "#91218c",
  // Purplish red
  "#e8e948",
  // Greenish yellow
  "#7e1510",
  // Reddish brown
  "#92ae31",
  // Yellow green
  "#6f340d",
  // Yellowish brown
  "#d32b1e",
  // Reddish orange
  "#2b3514"
  // Olive green
];
var SPECTRUM_12 = [
  "#4148cc",
  "#db3c80",
  "#12b5b0",
  "#ff8c14",
  "#848aff",
  "#78e16e",
  "#1e78f0",
  "#ebcd00",
  "#beeb3c",
  "#7828d2",
  "#cd5f00",
  "#00915f"
];
var CATEGORICAL_PALETTES = {
  tycho11: TYCHO_11,
  "tycho-dark11": TYCHO_DARK_11,
  "tycho-robust11": TYCHO_ROBUST_11,
  "tycho-soft11": TYCHO_SOFT_11,
  "tycho-soft-dark11": TYCHO_SOFT_DARK_11,
  "tycho-bold11": TYCHO_BOLD_11,
  "tycho-bold-dark11": TYCHO_BOLD_DARK_11,
  tableau10: TABLEAU_10,
  kelly22: KELLY_22,
  mathematica10: MATHEMATICA_10,
  cupertino10: CUPERTINO_10,
  "cupertino-dark10": CUPERTINO_DARK_10,
  spectrum12: SPECTRUM_12
};

// src/palettes/diverging.ts
var ROMA = [
  "#033198",
  "#053399",
  "#07359a",
  "#09379b",
  "#0b399c",
  "#0c3b9c",
  "#0e3d9d",
  "#0f3f9e",
  "#11409f",
  "#1242a0",
  "#1344a0",
  "#1446a1",
  "#1548a2",
  "#164aa3",
  "#174ca4",
  "#184da4",
  "#194fa5",
  "#1951a6",
  "#1a53a7",
  "#1b54a8",
  "#1b56a8",
  "#1c58a9",
  "#1d5aaa",
  "#1d5bab",
  "#1e5dab",
  "#1e5fac",
  "#1f60ad",
  "#2062ae",
  "#2064ae",
  "#2166af",
  "#2167b0",
  "#2269b0",
  "#226ab1",
  "#236cb2",
  "#236eb3",
  "#246fb3",
  "#2471b4",
  "#2573b5",
  "#2574b5",
  "#2676b6",
  "#2677b7",
  "#2779b7",
  "#287ab8",
  "#287cb9",
  "#297eb9",
  "#297fba",
  "#2a81bb",
  "#2b82bb",
  "#2b84bc",
  "#2c85bd",
  "#2d87bd",
  "#2d88be",
  "#2e8abf",
  "#2f8cbf",
  "#308dc0",
  "#308fc1",
  "#3190c1",
  "#3292c2",
  "#3394c3",
  "#3495c3",
  "#3597c4",
  "#3698c5",
  "#379ac5",
  "#389cc6",
  "#399dc7",
  "#3a9fc7",
  "#3ca1c8",
  "#3da2c9",
  "#3ea4c9",
  "#40a6ca",
  "#41a7cb",
  "#43a9cb",
  "#44abcc",
  "#46accc",
  "#47aecd",
  "#49b0ce",
  "#4bb2ce",
  "#4db3cf",
  "#4fb5d0",
  "#51b7d0",
  "#53b8d1",
  "#55bad1",
  "#57bcd2",
  "#59bdd2",
  "#5bbfd3",
  "#5dc1d3",
  "#60c3d4",
  "#62c4d4",
  "#64c6d5",
  "#67c7d5",
  "#69c9d5",
  "#6ccbd6",
  "#6eccd6",
  "#71ced6",
  "#74cfd6",
  "#76d1d7",
  "#79d2d7",
  "#7bd4d7",
  "#7ed5d7",
  "#81d7d7",
  "#83d8d7",
  "#86d9d7",
  "#89dad7",
  "#8bdcd7",
  "#8eddd7",
  "#90ded7",
  "#93dfd6",
  "#95e0d6",
  "#98e1d6",
  "#9be2d5",
  "#9de3d5",
  "#9fe4d4",
  "#a2e5d4",
  "#a4e5d3",
  "#a6e6d2",
  "#a9e7d2",
  "#abe7d1",
  "#ade8d0",
  "#afe8cf",
  "#b1e9ce",
  "#b3e9cd",
  "#b5eacc",
  "#b7eaca",
  "#b9eac9",
  "#bbeac8",
  "#bdeac6",
  "#beeac5",
  "#c0eac3",
  "#c1eac2",
  "#c3eac0",
  "#c4eabe",
  "#c6eabc",
  "#c7e9ba",
  "#c8e9b8",
  "#c9e9b6",
  "#cbe8b4",
  "#cce7b2",
  "#cce7b0",
  "#cde6ad",
  "#cee5ab",
  "#cfe5a8",
  "#d0e4a6",
  "#d0e3a3",
  "#d1e2a1",
  "#d1e19e",
  "#d1df9b",
  "#d2de98",
  "#d2dd96",
  "#d2dc93",
  "#d2da90",
  "#d2d98d",
  "#d2d78a",
  "#d2d687",
  "#d2d484",
  "#d2d381",
  "#d2d17e",
  "#d1cf7b",
  "#d1cd78",
  "#d0cc75",
  "#d0ca72",
  "#cfc86f",
  "#cfc66d",
  "#cec46a",
  "#cec267",
  "#cdc064",
  "#ccbe62",
  "#cbbc5f",
  "#cbba5d",
  "#cab85a",
  "#c9b658",
  "#c8b455",
  "#c7b253",
  "#c7b051",
  "#c6ae4f",
  "#c5ac4c",
  "#c4aa4a",
  "#c3a848",
  "#c2a647",
  "#c1a545",
  "#c1a343",
  "#c0a141",
  "#bf9f40",
  "#be9d3e",
  "#bd9b3c",
  "#bc993b",
  "#bb9739",
  "#ba9538",
  "#ba9437",
  "#b99235",
  "#b89034",
  "#b78e33",
  "#b68c32",
  "#b58b31",
  "#b58930",
  "#b4872f",
  "#b3862e",
  "#b2842d",
  "#b1822c",
  "#b0802b",
  "#b07f2a",
  "#af7d29",
  "#ae7c28",
  "#ad7a27",
  "#ad7826",
  "#ac7726",
  "#ab7525",
  "#aa7324",
  "#a97223",
  "#a97023",
  "#a86f22",
  "#a76d21",
  "#a66c20",
  "#a66a20",
  "#a5681f",
  "#a4671e",
  "#a3651e",
  "#a2641d",
  "#a2621c",
  "#a1611c",
  "#a05f1b",
  "#9f5d1a",
  "#9e5c19",
  "#9e5a19",
  "#9d5818",
  "#9c5717",
  "#9b5517",
  "#9a5316",
  "#995215",
  "#995014",
  "#984e14",
  "#974d13",
  "#964b12",
  "#954912",
  "#944711",
  "#934610",
  "#92440f",
  "#91420f",
  "#90400e",
  "#903e0d",
  "#8f3c0c",
  "#8e3a0b",
  "#8d380b",
  "#8c360a",
  "#8b3409",
  "#8a3208",
  "#883008",
  "#872e07",
  "#862c06",
  "#852a06",
  "#842705",
  "#832504",
  "#822204",
  "#812003",
  "#801d02",
  "#7f1a01",
  "#7e1700"
];
var VIK = [
  "#0b3c78",
  "#2366a8",
  "#4f97cf",
  "#9bc7e4",
  "#f7f7f7",
  "#e88fa0",
  "#c23c6f",
  "#8e1d4d",
  "#5a0c2e"
];
var BROC = [
  "#2c557f",
  "#4a7fa8",
  "#7aa8c9",
  "#b5cfe3",
  "#f7f7f7",
  "#e4b0b0",
  "#c96c72",
  "#9b3d4f",
  "#6b1a2b"
];
var RDBU = [
  "#053061",
  "#11458c",
  "#1f5fa5",
  "#337ab7",
  "#4f94c6",
  "#6baed6",
  "#8bbdd9",
  "#a6cce3",
  "#c1dbec",
  "#dceaf4",
  "#f7f7f7",
  "#fbe3d6",
  "#f9c8b2",
  "#f5a993",
  "#ee8a75",
  "#e36b58",
  "#d64b41",
  "#c72c33",
  "#af212d",
  "#971628",
  "#7f0b23",
  "#67001f"
];
var COOLWARM = [
  "#3b4cc0",
  "#465ecf",
  "#5070de",
  "#5c82ea",
  "#6f96f5",
  "#86aafc",
  "#9dbdff",
  "#b6d0f3",
  "#cfe1e8",
  "#e6efe1",
  "#f1f1f1",
  "#f7f7f7",
  "#f6e1d3",
  "#f4cbb3",
  "#f1b594",
  "#ee9f76",
  "#ea8960",
  "#e06b53",
  "#d24f49",
  "#c1363f",
  "#ae2036",
  "#b40426"
];
var reversePalette = (palette) => palette.slice().reverse();
var DIVERGING_PALETTES = {
  roma: ROMA,
  "roma-reversed": reversePalette(ROMA),
  vik: VIK,
  "vik-reversed": reversePalette(VIK),
  broc: BROC,
  "broc-reversed": reversePalette(BROC),
  rdbu: RDBU,
  "rdbu-reversed": reversePalette(RDBU),
  coolwarm: COOLWARM,
  "coolwarm-reversed": reversePalette(COOLWARM)
};

// src/palettes/sequential.ts
var TURBO = [
  "#30123b",
  "#321543",
  "#33184a",
  "#341b51",
  "#351e58",
  "#36215f",
  "#372466",
  "#38276d",
  "#392a73",
  "#3a2d79",
  "#3b2f80",
  "#3c3286",
  "#3d358b",
  "#3e3891",
  "#3f3b97",
  "#3f3e9c",
  "#4040a2",
  "#4143a7",
  "#4146ac",
  "#4249b1",
  "#424bb5",
  "#434eba",
  "#4451bf",
  "#4454c3",
  "#4456c7",
  "#4559cb",
  "#455ccf",
  "#455ed3",
  "#4661d6",
  "#4664da",
  "#4666dd",
  "#4669e0",
  "#466be3",
  "#476ee6",
  "#4771e9",
  "#4773eb",
  "#4776ee",
  "#4778f0",
  "#477bf2",
  "#467df4",
  "#4680f6",
  "#4682f8",
  "#4685fa",
  "#4687fb",
  "#458afc",
  "#458cfd",
  "#448ffe",
  "#4391fe",
  "#4294ff",
  "#4196ff",
  "#4099ff",
  "#3e9bfe",
  "#3d9efe",
  "#3ba0fd",
  "#3aa3fc",
  "#38a5fb",
  "#37a8fa",
  "#35abf8",
  "#33adf7",
  "#31aff5",
  "#2fb2f4",
  "#2eb4f2",
  "#2cb7f0",
  "#2ab9ee",
  "#28bceb",
  "#27bee9",
  "#25c0e7",
  "#23c3e4",
  "#22c5e2",
  "#20c7df",
  "#1fc9dd",
  "#1ecbda",
  "#1ccdd8",
  "#1bd0d5",
  "#1ad2d2",
  "#1ad4d0",
  "#19d5cd",
  "#18d7ca",
  "#18d9c8",
  "#18dbc5",
  "#18ddc2",
  "#18dec0",
  "#18e0bd",
  "#19e2bb",
  "#19e3b9",
  "#1ae4b6",
  "#1ce6b4",
  "#1de7b2",
  "#1fe9af",
  "#20eaac",
  "#22ebaa",
  "#25eca7",
  "#27eea4",
  "#2aefa1",
  "#2cf09e",
  "#2ff19b",
  "#32f298",
  "#35f394",
  "#38f491",
  "#3cf58e",
  "#3ff68a",
  "#43f787",
  "#46f884",
  "#4af880",
  "#4ef97d",
  "#52fa7a",
  "#55fa76",
  "#59fb73",
  "#5dfc6f",
  "#61fc6c",
  "#65fd69",
  "#69fd66",
  "#6dfe62",
  "#71fe5f",
  "#75fe5c",
  "#79fe59",
  "#7dff56",
  "#80ff53",
  "#84ff51",
  "#88ff4e",
  "#8bff4b",
  "#8fff49",
  "#92ff47",
  "#96fe44",
  "#99fe42",
  "#9cfe40",
  "#9ffd3f",
  "#a1fd3d",
  "#a4fc3c",
  "#a7fc3a",
  "#a9fb39",
  "#acfb38",
  "#affa37",
  "#b1f936",
  "#b4f836",
  "#b7f735",
  "#b9f635",
  "#bcf534",
  "#bef434",
  "#c1f334",
  "#c3f134",
  "#c6f034",
  "#c8ef34",
  "#cbed34",
  "#cdec34",
  "#d0ea34",
  "#d2e935",
  "#d4e735",
  "#d7e535",
  "#d9e436",
  "#dbe236",
  "#dde037",
  "#dfdf37",
  "#e1dd37",
  "#e3db38",
  "#e5d938",
  "#e7d739",
  "#e9d539",
  "#ebd339",
  "#ecd13a",
  "#eecf3a",
  "#efcd3a",
  "#f1cb3a",
  "#f2c93a",
  "#f4c73a",
  "#f5c53a",
  "#f6c33a",
  "#f7c13a",
  "#f8be39",
  "#f9bc39",
  "#faba39",
  "#fbb838",
  "#fbb637",
  "#fcb336",
  "#fcb136",
  "#fdae35",
  "#fdac34",
  "#fea933",
  "#fea732",
  "#fea431",
  "#fea130",
  "#fe9e2f",
  "#fe9b2d",
  "#fe992c",
  "#fe962b",
  "#fe932a",
  "#fe9029",
  "#fd8d27",
  "#fd8a26",
  "#fc8725",
  "#fc8423",
  "#fb8122",
  "#fb7e21",
  "#fa7b1f",
  "#f9781e",
  "#f9751d",
  "#f8721c",
  "#f76f1a",
  "#f66c19",
  "#f56918",
  "#f46617",
  "#f36315",
  "#f26014",
  "#f15d13",
  "#f05b12",
  "#ef5811",
  "#ed5510",
  "#ec530f",
  "#eb500e",
  "#ea4e0d",
  "#e84b0c",
  "#e7490c",
  "#e5470b",
  "#e4450a",
  "#e2430a",
  "#e14109",
  "#df3f08",
  "#dd3d08",
  "#dc3b07",
  "#da3907",
  "#d83706",
  "#d63506",
  "#d43305",
  "#d23105",
  "#d02f05",
  "#ce2d04",
  "#cc2b04",
  "#ca2a04",
  "#c82803",
  "#c52603",
  "#c32503",
  "#c12302",
  "#be2102",
  "#bc2002",
  "#b91e02",
  "#b71d02",
  "#b41b01",
  "#b21a01",
  "#af1801",
  "#ac1701",
  "#a91601",
  "#a71401",
  "#a41301",
  "#a11201",
  "#9e1001",
  "#9b0f01",
  "#980e01",
  "#950d01",
  "#920b01",
  "#8e0a01",
  "#8b0902",
  "#880802",
  "#850702",
  "#810602",
  "#7e0502",
  "#7a0403"
];
var MAGMA = [
  "#000004",
  "#010005",
  "#010106",
  "#010108",
  "#020109",
  "#02020b",
  "#02020d",
  "#03030f",
  "#030312",
  "#040414",
  "#050416",
  "#060518",
  "#06051a",
  "#07061c",
  "#08071e",
  "#090720",
  "#0a0822",
  "#0b0924",
  "#0c0926",
  "#0d0a29",
  "#0e0b2b",
  "#100b2d",
  "#110c2f",
  "#120d31",
  "#130d34",
  "#140e36",
  "#150e38",
  "#160f3b",
  "#180f3d",
  "#19103f",
  "#1a1042",
  "#1c1044",
  "#1d1147",
  "#1e1149",
  "#20114b",
  "#21114e",
  "#221150",
  "#241253",
  "#251255",
  "#271258",
  "#29115a",
  "#2a115c",
  "#2c115f",
  "#2d1161",
  "#2f1163",
  "#311165",
  "#331067",
  "#341069",
  "#36106b",
  "#38106c",
  "#390f6e",
  "#3b0f70",
  "#3d0f71",
  "#3f0f72",
  "#400f74",
  "#420f75",
  "#440f76",
  "#451077",
  "#471078",
  "#491078",
  "#4a1079",
  "#4c117a",
  "#4e117b",
  "#4f127b",
  "#51127c",
  "#52137c",
  "#54137d",
  "#56147d",
  "#57157e",
  "#59157e",
  "#5a167e",
  "#5c167f",
  "#5d177f",
  "#5f187f",
  "#601880",
  "#621980",
  "#641a80",
  "#651a80",
  "#671b80",
  "#681c81",
  "#6a1c81",
  "#6b1d81",
  "#6d1d81",
  "#6e1e81",
  "#701f81",
  "#721f81",
  "#732081",
  "#752181",
  "#762181",
  "#782281",
  "#792282",
  "#7b2382",
  "#7c2382",
  "#7e2482",
  "#802582",
  "#812581",
  "#832681",
  "#842681",
  "#862781",
  "#882781",
  "#892881",
  "#8b2981",
  "#8c2981",
  "#8e2a81",
  "#902a81",
  "#912b81",
  "#932b80",
  "#942c80",
  "#962c80",
  "#982d80",
  "#992d80",
  "#9b2e7f",
  "#9c2e7f",
  "#9e2f7f",
  "#a02f7f",
  "#a1307e",
  "#a3307e",
  "#a5317e",
  "#a6317d",
  "#a8327d",
  "#aa337d",
  "#ab337c",
  "#ad347c",
  "#ae347b",
  "#b0357b",
  "#b2357b",
  "#b3367a",
  "#b5367a",
  "#b73779",
  "#b83779",
  "#ba3878",
  "#bc3978",
  "#bd3977",
  "#bf3a77",
  "#c03a76",
  "#c23b75",
  "#c43c75",
  "#c53c74",
  "#c73d73",
  "#c83e73",
  "#ca3e72",
  "#cc3f71",
  "#cd4071",
  "#cf4070",
  "#d0416f",
  "#d2426f",
  "#d3436e",
  "#d5446d",
  "#d6456c",
  "#d8456c",
  "#d9466b",
  "#db476a",
  "#dc4869",
  "#de4968",
  "#df4a68",
  "#e04c67",
  "#e24d66",
  "#e34e65",
  "#e44f64",
  "#e55064",
  "#e75263",
  "#e85362",
  "#e95462",
  "#ea5661",
  "#eb5760",
  "#ec5860",
  "#ed5a5f",
  "#ee5b5e",
  "#ef5d5e",
  "#f05f5e",
  "#f1605d",
  "#f2625d",
  "#f2645c",
  "#f3655c",
  "#f4675c",
  "#f4695c",
  "#f56b5c",
  "#f66c5c",
  "#f66e5c",
  "#f7705c",
  "#f7725c",
  "#f8745c",
  "#f8765c",
  "#f9785d",
  "#f9795d",
  "#f97b5d",
  "#fa7d5e",
  "#fa7f5e",
  "#fa815f",
  "#fb835f",
  "#fb8560",
  "#fb8761",
  "#fc8961",
  "#fc8a62",
  "#fc8c63",
  "#fc8e64",
  "#fc9065",
  "#fd9266",
  "#fd9467",
  "#fd9668",
  "#fd9869",
  "#fd9a6a",
  "#fd9b6b",
  "#fe9d6c",
  "#fe9f6d",
  "#fea16e",
  "#fea36f",
  "#fea571",
  "#fea772",
  "#fea973",
  "#feaa74",
  "#feac76",
  "#feae77",
  "#feb078",
  "#feb27a",
  "#feb47b",
  "#feb67c",
  "#feb77e",
  "#feb97f",
  "#febb81",
  "#febd82",
  "#febf84",
  "#fec185",
  "#fec287",
  "#fec488",
  "#fec68a",
  "#fec88c",
  "#feca8d",
  "#fecc8f",
  "#fecd90",
  "#fecf92",
  "#fed194",
  "#fed395",
  "#fed597",
  "#fed799",
  "#fed89a",
  "#fdda9c",
  "#fddc9e",
  "#fddea0",
  "#fde0a1",
  "#fde2a3",
  "#fde3a5",
  "#fde5a7",
  "#fde7a9",
  "#fde9aa",
  "#fdebac",
  "#fcecae",
  "#fceeb0",
  "#fcf0b2",
  "#fcf2b4",
  "#fcf4b6",
  "#fcf6b8",
  "#fcf7b9",
  "#fcf9bb",
  "#fcfbbd",
  "#fcfdbf"
];
var INFERNO = [
  "#000004",
  "#010005",
  "#010106",
  "#010108",
  "#02010a",
  "#02020c",
  "#02020e",
  "#030210",
  "#040312",
  "#040314",
  "#050417",
  "#060419",
  "#07051b",
  "#08051d",
  "#09061f",
  "#0a0722",
  "#0b0724",
  "#0c0826",
  "#0d0829",
  "#0e092b",
  "#10092d",
  "#110a30",
  "#120a32",
  "#140b34",
  "#150b37",
  "#160b39",
  "#180c3c",
  "#190c3e",
  "#1b0c41",
  "#1c0c43",
  "#1e0c45",
  "#1f0c48",
  "#210c4a",
  "#230c4c",
  "#240c4f",
  "#260c51",
  "#280b53",
  "#290b55",
  "#2b0b57",
  "#2d0b59",
  "#2f0a5b",
  "#310a5c",
  "#320a5e",
  "#340a5f",
  "#360961",
  "#380962",
  "#390963",
  "#3b0964",
  "#3d0965",
  "#3e0966",
  "#400a67",
  "#420a68",
  "#440a68",
  "#450a69",
  "#470b6a",
  "#490b6a",
  "#4a0c6b",
  "#4c0c6b",
  "#4d0d6c",
  "#4f0d6c",
  "#510e6c",
  "#520e6d",
  "#540f6d",
  "#550f6d",
  "#57106e",
  "#59106e",
  "#5a116e",
  "#5c126e",
  "#5d126e",
  "#5f136e",
  "#61136e",
  "#62146e",
  "#64156e",
  "#65156e",
  "#67166e",
  "#69166e",
  "#6a176e",
  "#6c186e",
  "#6d186e",
  "#6f196e",
  "#71196e",
  "#721a6e",
  "#741a6e",
  "#751b6e",
  "#771c6d",
  "#781c6d",
  "#7a1d6d",
  "#7c1d6d",
  "#7d1e6d",
  "#7f1e6c",
  "#801f6c",
  "#82206c",
  "#84206b",
  "#85216b",
  "#87216b",
  "#88226a",
  "#8a226a",
  "#8c2369",
  "#8d2369",
  "#8f2469",
  "#902568",
  "#922568",
  "#932667",
  "#952667",
  "#972766",
  "#982766",
  "#9a2865",
  "#9b2964",
  "#9d2964",
  "#9f2a63",
  "#a02a63",
  "#a22b62",
  "#a32c61",
  "#a52c60",
  "#a62d60",
  "#a82e5f",
  "#a92e5e",
  "#ab2f5e",
  "#ad305d",
  "#ae305c",
  "#b0315b",
  "#b1325a",
  "#b3325a",
  "#b43359",
  "#b63458",
  "#b73557",
  "#b93556",
  "#ba3655",
  "#bc3754",
  "#bd3853",
  "#bf3952",
  "#c03a51",
  "#c13a50",
  "#c33b4f",
  "#c43c4e",
  "#c63d4d",
  "#c73e4c",
  "#c83f4b",
  "#ca404a",
  "#cb4149",
  "#cc4248",
  "#ce4347",
  "#cf4446",
  "#d04545",
  "#d24644",
  "#d34743",
  "#d44842",
  "#d54a41",
  "#d74b3f",
  "#d84c3e",
  "#d94d3d",
  "#da4e3c",
  "#db503b",
  "#dd513a",
  "#de5238",
  "#df5337",
  "#e05536",
  "#e15635",
  "#e25734",
  "#e35933",
  "#e45a31",
  "#e55c30",
  "#e65d2f",
  "#e75e2e",
  "#e8602d",
  "#e9612b",
  "#ea632a",
  "#eb6429",
  "#eb6628",
  "#ec6726",
  "#ed6925",
  "#ee6a24",
  "#ef6c23",
  "#ef6e21",
  "#f06f20",
  "#f1711f",
  "#f1731d",
  "#f2741c",
  "#f3761b",
  "#f37819",
  "#f47918",
  "#f57b17",
  "#f57d15",
  "#f67e14",
  "#f68013",
  "#f78212",
  "#f78410",
  "#f8850f",
  "#f8870e",
  "#f8890c",
  "#f98b0b",
  "#f98c0a",
  "#f98e09",
  "#fa9008",
  "#fa9207",
  "#fa9407",
  "#fb9606",
  "#fb9706",
  "#fb9906",
  "#fb9b06",
  "#fb9d07",
  "#fc9f07",
  "#fca108",
  "#fca309",
  "#fca50a",
  "#fca60c",
  "#fca80d",
  "#fcaa0f",
  "#fcac11",
  "#fcae12",
  "#fcb014",
  "#fcb216",
  "#fcb418",
  "#fbb61a",
  "#fbb81d",
  "#fbba1f",
  "#fbbc21",
  "#fbbe23",
  "#fac026",
  "#fac228",
  "#fac42a",
  "#fac62d",
  "#f9c72f",
  "#f9c932",
  "#f9cb35",
  "#f8cd37",
  "#f8cf3a",
  "#f7d13d",
  "#f7d340",
  "#f6d543",
  "#f6d746",
  "#f5d949",
  "#f5db4c",
  "#f4dd4f",
  "#f4df53",
  "#f4e156",
  "#f3e35a",
  "#f3e55d",
  "#f2e661",
  "#f2e865",
  "#f2ea69",
  "#f1ec6d",
  "#f1ed71",
  "#f1ef75",
  "#f1f179",
  "#f2f27d",
  "#f2f482",
  "#f3f586",
  "#f3f68a",
  "#f4f88e",
  "#f5f992",
  "#f6fa96",
  "#f8fb9a",
  "#f9fc9d",
  "#fafda1",
  "#fcffa4"
];
var PLASMA = [
  "#0d0887",
  "#100788",
  "#130789",
  "#16078a",
  "#19068c",
  "#1b068d",
  "#1d068e",
  "#20068f",
  "#220690",
  "#240691",
  "#260591",
  "#280592",
  "#2a0593",
  "#2c0594",
  "#2e0595",
  "#2f0596",
  "#310597",
  "#330597",
  "#350498",
  "#370499",
  "#38049a",
  "#3a049a",
  "#3c049b",
  "#3e049c",
  "#3f049c",
  "#41049d",
  "#43039e",
  "#44039e",
  "#46039f",
  "#48039f",
  "#4903a0",
  "#4b03a1",
  "#4c02a1",
  "#4e02a2",
  "#5002a2",
  "#5102a3",
  "#5302a3",
  "#5502a4",
  "#5601a4",
  "#5801a4",
  "#5901a5",
  "#5b01a5",
  "#5c01a6",
  "#5e01a6",
  "#6001a6",
  "#6100a7",
  "#6300a7",
  "#6400a7",
  "#6600a7",
  "#6700a8",
  "#6900a8",
  "#6a00a8",
  "#6c00a8",
  "#6e00a8",
  "#6f00a8",
  "#7100a8",
  "#7201a8",
  "#7401a8",
  "#7501a8",
  "#7701a8",
  "#7801a8",
  "#7a02a8",
  "#7b02a8",
  "#7d03a8",
  "#7e03a8",
  "#8004a8",
  "#8104a7",
  "#8305a7",
  "#8405a7",
  "#8606a6",
  "#8707a6",
  "#8808a6",
  "#8a09a5",
  "#8b0aa5",
  "#8d0ba5",
  "#8e0ca4",
  "#8f0da4",
  "#910ea3",
  "#920fa3",
  "#9410a2",
  "#9511a1",
  "#9613a1",
  "#9814a0",
  "#99159f",
  "#9a169f",
  "#9c179e",
  "#9d189d",
  "#9e199d",
  "#a01a9c",
  "#a11b9b",
  "#a21d9a",
  "#a31e9a",
  "#a51f99",
  "#a62098",
  "#a72197",
  "#a82296",
  "#aa2395",
  "#ab2494",
  "#ac2694",
  "#ad2793",
  "#ae2892",
  "#b02991",
  "#b12a90",
  "#b22b8f",
  "#b32c8e",
  "#b42e8d",
  "#b52f8c",
  "#b6308b",
  "#b7318a",
  "#b83289",
  "#ba3388",
  "#bb3488",
  "#bc3587",
  "#bd3786",
  "#be3885",
  "#bf3984",
  "#c03a83",
  "#c13b82",
  "#c23c81",
  "#c33d80",
  "#c43e7f",
  "#c5407e",
  "#c6417d",
  "#c7427c",
  "#c8437b",
  "#c9447a",
  "#ca457a",
  "#cb4679",
  "#cc4778",
  "#cc4977",
  "#cd4a76",
  "#ce4b75",
  "#cf4c74",
  "#d04d73",
  "#d14e72",
  "#d24f71",
  "#d35171",
  "#d45270",
  "#d5536f",
  "#d5546e",
  "#d6556d",
  "#d7566c",
  "#d8576b",
  "#d9586a",
  "#da5a6a",
  "#da5b69",
  "#db5c68",
  "#dc5d67",
  "#dd5e66",
  "#de5f65",
  "#de6164",
  "#df6263",
  "#e06363",
  "#e16462",
  "#e26561",
  "#e26660",
  "#e3685f",
  "#e4695e",
  "#e56a5d",
  "#e56b5d",
  "#e66c5c",
  "#e76e5b",
  "#e76f5a",
  "#e87059",
  "#e97158",
  "#e97257",
  "#ea7457",
  "#eb7556",
  "#eb7655",
  "#ec7754",
  "#ed7953",
  "#ed7a52",
  "#ee7b51",
  "#ef7c51",
  "#ef7e50",
  "#f07f4f",
  "#f0804e",
  "#f1814d",
  "#f1834c",
  "#f2844b",
  "#f3854b",
  "#f3874a",
  "#f48849",
  "#f48948",
  "#f58b47",
  "#f58c46",
  "#f68d45",
  "#f68f44",
  "#f79044",
  "#f79143",
  "#f79342",
  "#f89441",
  "#f89540",
  "#f9973f",
  "#f9983e",
  "#f99a3e",
  "#fa9b3d",
  "#fa9c3c",
  "#fa9e3b",
  "#fb9f3a",
  "#fba139",
  "#fba238",
  "#fca338",
  "#fca537",
  "#fca636",
  "#fca835",
  "#fca934",
  "#fdab33",
  "#fdac33",
  "#fdae32",
  "#fdaf31",
  "#fdb130",
  "#fdb22f",
  "#fdb42f",
  "#fdb52e",
  "#feb72d",
  "#feb82c",
  "#feba2c",
  "#febb2b",
  "#febd2a",
  "#febe2a",
  "#fec029",
  "#fdc229",
  "#fdc328",
  "#fdc527",
  "#fdc627",
  "#fdc827",
  "#fdca26",
  "#fdcb26",
  "#fccd25",
  "#fcce25",
  "#fcd025",
  "#fcd225",
  "#fbd324",
  "#fbd524",
  "#fbd724",
  "#fad824",
  "#fada24",
  "#f9dc24",
  "#f9dd25",
  "#f8df25",
  "#f8e125",
  "#f7e225",
  "#f7e425",
  "#f6e626",
  "#f6e826",
  "#f5e926",
  "#f5eb27",
  "#f4ed27",
  "#f3ee27",
  "#f3f027",
  "#f2f227",
  "#f1f426",
  "#f1f525",
  "#f0f724",
  "#f0f921"
];
var VIRIDIS = [
  "#440154",
  "#440256",
  "#450457",
  "#450559",
  "#46075a",
  "#46085c",
  "#460a5d",
  "#460b5e",
  "#470d60",
  "#470e61",
  "#471063",
  "#471164",
  "#471365",
  "#481467",
  "#481668",
  "#481769",
  "#48186a",
  "#481a6c",
  "#481b6d",
  "#481c6e",
  "#481d6f",
  "#481f70",
  "#482071",
  "#482173",
  "#482374",
  "#482475",
  "#482576",
  "#482677",
  "#482878",
  "#482979",
  "#472a7a",
  "#472c7a",
  "#472d7b",
  "#472e7c",
  "#472f7d",
  "#46307e",
  "#46327e",
  "#46337f",
  "#463480",
  "#453581",
  "#453781",
  "#453882",
  "#443983",
  "#443a83",
  "#443b84",
  "#433d84",
  "#433e85",
  "#423f85",
  "#424086",
  "#424186",
  "#414287",
  "#414487",
  "#404588",
  "#404688",
  "#3f4788",
  "#3f4889",
  "#3e4989",
  "#3e4a89",
  "#3e4c8a",
  "#3d4d8a",
  "#3d4e8a",
  "#3c4f8a",
  "#3c508b",
  "#3b518b",
  "#3b528b",
  "#3a538b",
  "#3a548c",
  "#39558c",
  "#39568c",
  "#38588c",
  "#38598c",
  "#375a8c",
  "#375b8d",
  "#365c8d",
  "#365d8d",
  "#355e8d",
  "#355f8d",
  "#34608d",
  "#34618d",
  "#33628d",
  "#33638d",
  "#32648e",
  "#32658e",
  "#31668e",
  "#31678e",
  "#31688e",
  "#30698e",
  "#306a8e",
  "#2f6b8e",
  "#2f6c8e",
  "#2e6d8e",
  "#2e6e8e",
  "#2e6f8e",
  "#2d708e",
  "#2d718e",
  "#2c718e",
  "#2c728e",
  "#2c738e",
  "#2b748e",
  "#2b758e",
  "#2a768e",
  "#2a778e",
  "#2a788e",
  "#29798e",
  "#297a8e",
  "#297b8e",
  "#287c8e",
  "#287d8e",
  "#277e8e",
  "#277f8e",
  "#27808e",
  "#26818e",
  "#26828e",
  "#26828e",
  "#25838e",
  "#25848e",
  "#25858e",
  "#24868e",
  "#24878e",
  "#23888e",
  "#23898e",
  "#238a8d",
  "#228b8d",
  "#228c8d",
  "#228d8d",
  "#218e8d",
  "#218f8d",
  "#21908d",
  "#21918c",
  "#20928c",
  "#20928c",
  "#20938c",
  "#1f948c",
  "#1f958b",
  "#1f968b",
  "#1f978b",
  "#1f988b",
  "#1f998a",
  "#1f9a8a",
  "#1e9b8a",
  "#1e9c89",
  "#1e9d89",
  "#1f9e89",
  "#1f9f88",
  "#1fa088",
  "#1fa188",
  "#1fa187",
  "#1fa287",
  "#20a386",
  "#20a486",
  "#21a585",
  "#21a685",
  "#22a785",
  "#22a884",
  "#23a983",
  "#24aa83",
  "#25ab82",
  "#25ac82",
  "#26ad81",
  "#27ad81",
  "#28ae80",
  "#29af7f",
  "#2ab07f",
  "#2cb17e",
  "#2db27d",
  "#2eb37c",
  "#2fb47c",
  "#31b57b",
  "#32b67a",
  "#34b679",
  "#35b779",
  "#37b878",
  "#38b977",
  "#3aba76",
  "#3bbb75",
  "#3dbc74",
  "#3fbc73",
  "#40bd72",
  "#42be71",
  "#44bf70",
  "#46c06f",
  "#48c16e",
  "#4ac16d",
  "#4cc26c",
  "#4ec36b",
  "#50c46a",
  "#52c569",
  "#54c568",
  "#56c667",
  "#58c765",
  "#5ac864",
  "#5cc863",
  "#5ec962",
  "#60ca60",
  "#63cb5f",
  "#65cb5e",
  "#67cc5c",
  "#69cd5b",
  "#6ccd5a",
  "#6ece58",
  "#70cf57",
  "#73d056",
  "#75d054",
  "#77d153",
  "#7ad151",
  "#7cd250",
  "#7fd34e",
  "#81d34d",
  "#84d44b",
  "#86d549",
  "#89d548",
  "#8bd646",
  "#8ed645",
  "#90d743",
  "#93d741",
  "#95d840",
  "#98d83e",
  "#9bd93c",
  "#9dd93b",
  "#a0da39",
  "#a2da37",
  "#a5db36",
  "#a8db34",
  "#aadc32",
  "#addc30",
  "#b0dd2f",
  "#b2dd2d",
  "#b5de2b",
  "#b8de29",
  "#bade28",
  "#bddf26",
  "#c0df25",
  "#c2df23",
  "#c5e021",
  "#c8e020",
  "#cae11f",
  "#cde11d",
  "#d0e11c",
  "#d2e21b",
  "#d5e21a",
  "#d8e219",
  "#dae319",
  "#dde318",
  "#dfe318",
  "#e2e418",
  "#e5e419",
  "#e7e419",
  "#eae51a",
  "#ece51b",
  "#efe51c",
  "#f1e51d",
  "#f4e61e",
  "#f6e620",
  "#f8e621",
  "#fbe723",
  "#fde725"
];
var CIVIDIS = [
  "#00204d",
  "#00214e",
  "#002250",
  "#002252",
  "#002353",
  "#002455",
  "#002557",
  "#002558",
  "#00265a",
  "#00275c",
  "#00275e",
  "#002860",
  "#002961",
  "#002a63",
  "#002a65",
  "#002b67",
  "#002c69",
  "#002c6a",
  "#002d6c",
  "#002e6e",
  "#002e6f",
  "#002f6f",
  "#002f6f",
  "#00306f",
  "#00306f",
  "#00316f",
  "#00326f",
  "#00336f",
  "#00336f",
  "#00346f",
  "#00356e",
  "#01366e",
  "#06366e",
  "#0b376e",
  "#0f386e",
  "#12386d",
  "#15396d",
  "#183a6d",
  "#1a3b6d",
  "#1d3b6d",
  "#1f3c6d",
  "#213d6d",
  "#233e6c",
  "#243e6c",
  "#263f6c",
  "#28406c",
  "#2a406c",
  "#2b416c",
  "#2d426c",
  "#2e436c",
  "#30436c",
  "#31446b",
  "#32456b",
  "#34456b",
  "#35466b",
  "#36476b",
  "#38486b",
  "#39486b",
  "#3a496b",
  "#3b4a6b",
  "#3d4a6b",
  "#3e4b6b",
  "#3f4c6b",
  "#404d6b",
  "#414d6b",
  "#424e6b",
  "#434f6b",
  "#444f6b",
  "#46506b",
  "#47516b",
  "#48526b",
  "#49526b",
  "#4a536b",
  "#4b546c",
  "#4c546c",
  "#4d556c",
  "#4e566c",
  "#4f576c",
  "#50576c",
  "#51586c",
  "#52596c",
  "#53596c",
  "#545a6c",
  "#555b6d",
  "#565c6d",
  "#575c6d",
  "#585d6d",
  "#595e6d",
  "#595f6d",
  "#5a5f6d",
  "#5b606e",
  "#5c616e",
  "#5d616e",
  "#5e626e",
  "#5f636e",
  "#60646f",
  "#61646f",
  "#62656f",
  "#63666f",
  "#64666f",
  "#646770",
  "#656870",
  "#666970",
  "#676970",
  "#686a71",
  "#696b71",
  "#6a6c71",
  "#6b6c71",
  "#6c6d72",
  "#6c6e72",
  "#6d6e72",
  "#6e6f73",
  "#6f7073",
  "#707173",
  "#717174",
  "#727274",
  "#727374",
  "#737475",
  "#747475",
  "#757575",
  "#767676",
  "#777776",
  "#787777",
  "#787877",
  "#797977",
  "#7a7a78",
  "#7b7a78",
  "#7c7b78",
  "#7d7c78",
  "#7e7d78",
  "#7f7d78",
  "#807e79",
  "#817f79",
  "#828079",
  "#838079",
  "#848179",
  "#848279",
  "#858379",
  "#868379",
  "#878479",
  "#888579",
  "#898679",
  "#8a8779",
  "#8b8779",
  "#8c8879",
  "#8d8979",
  "#8e8a79",
  "#8f8a79",
  "#908b79",
  "#918c78",
  "#928d78",
  "#938e78",
  "#948e78",
  "#958f78",
  "#969078",
  "#979178",
  "#989278",
  "#999278",
  "#9a9377",
  "#9b9477",
  "#9c9577",
  "#9d9677",
  "#9e9677",
  "#9f9777",
  "#a09877",
  "#a19976",
  "#a29a76",
  "#a39a76",
  "#a49b76",
  "#a59c76",
  "#a69d75",
  "#a89e75",
  "#a99f75",
  "#aa9f75",
  "#aba074",
  "#aca174",
  "#ada274",
  "#aea374",
  "#afa473",
  "#b0a473",
  "#b1a573",
  "#b2a672",
  "#b3a772",
  "#b4a872",
  "#b5a971",
  "#b6a971",
  "#b7aa71",
  "#b8ab70",
  "#b9ac70",
  "#baad70",
  "#bbae6f",
  "#bcaf6f",
  "#beaf6f",
  "#bfb06e",
  "#c0b16e",
  "#c1b26d",
  "#c2b36d",
  "#c3b46d",
  "#c4b56c",
  "#c5b56c",
  "#c6b66b",
  "#c7b76b",
  "#c8b86a",
  "#c9b96a",
  "#cbba69",
  "#ccbb69",
  "#cdbc68",
  "#cebc68",
  "#cfbd67",
  "#d0be67",
  "#d1bf66",
  "#d2c066",
  "#d3c165",
  "#d4c264",
  "#d6c364",
  "#d7c463",
  "#d8c563",
  "#d9c562",
  "#dac661",
  "#dbc761",
  "#dcc860",
  "#ddc95f",
  "#deca5f",
  "#e0cb5e",
  "#e1cc5d",
  "#e2cd5c",
  "#e3ce5c",
  "#e4cf5b",
  "#e5d05a",
  "#e6d159",
  "#e8d259",
  "#e9d358",
  "#ead357",
  "#ebd456",
  "#ecd555",
  "#edd654",
  "#efd753",
  "#f0d852",
  "#f1d951",
  "#f2da50",
  "#f3db4f",
  "#f4dc4e",
  "#f6dd4d",
  "#f7de4c",
  "#f8df4b",
  "#f9e04a",
  "#fae149",
  "#fbe248",
  "#fde346",
  "#fee445",
  "#ffe544",
  "#ffe642",
  "#ffe742",
  "#ffe843",
  "#ffe944",
  "#ffea46"
];
var ROCKET = [
  "#03051a",
  "#04051a",
  "#05061b",
  "#06071c",
  "#07071d",
  "#08081e",
  "#0a091f",
  "#0b0920",
  "#0d0a21",
  "#0e0b22",
  "#100b23",
  "#110c24",
  "#130d25",
  "#140e26",
  "#160e27",
  "#170f28",
  "#180f29",
  "#1a102a",
  "#1b112b",
  "#1d112c",
  "#1e122d",
  "#20122e",
  "#211330",
  "#221331",
  "#241432",
  "#251433",
  "#271534",
  "#281535",
  "#2a1636",
  "#2b1637",
  "#2d1738",
  "#2e1739",
  "#30173a",
  "#31183b",
  "#33183c",
  "#34193d",
  "#35193e",
  "#37193f",
  "#381a40",
  "#3a1a41",
  "#3c1a42",
  "#3d1a42",
  "#3f1b43",
  "#401b44",
  "#421b45",
  "#431c46",
  "#451c47",
  "#461c48",
  "#481c48",
  "#491d49",
  "#4b1d4a",
  "#4c1d4b",
  "#4e1d4b",
  "#501d4c",
  "#511e4d",
  "#531e4d",
  "#541e4e",
  "#561e4f",
  "#581e4f",
  "#591e50",
  "#5b1e51",
  "#5c1e51",
  "#5e1f52",
  "#601f52",
  "#611f53",
  "#631f53",
  "#641f54",
  "#661f54",
  "#681f55",
  "#691f55",
  "#6b1f56",
  "#6d1f56",
  "#6e1f57",
  "#701f57",
  "#711f57",
  "#731f58",
  "#751f58",
  "#761f58",
  "#781f59",
  "#7a1f59",
  "#7b1f59",
  "#7d1f5a",
  "#7f1e5a",
  "#811e5a",
  "#821e5a",
  "#841e5a",
  "#861e5b",
  "#871e5b",
  "#891e5b",
  "#8b1d5b",
  "#8c1d5b",
  "#8e1d5b",
  "#901d5b",
  "#921c5b",
  "#931c5b",
  "#951c5b",
  "#971c5b",
  "#981b5b",
  "#9a1b5b",
  "#9c1b5b",
  "#9e1a5b",
  "#9f1a5b",
  "#a11a5b",
  "#a3195b",
  "#a4195b",
  "#a6195a",
  "#a8185a",
  "#aa185a",
  "#ab185a",
  "#ad1759",
  "#af1759",
  "#b01759",
  "#b21758",
  "#b41658",
  "#b51657",
  "#b71657",
  "#b91657",
  "#ba1656",
  "#bc1656",
  "#bd1655",
  "#bf1654",
  "#c11754",
  "#c21753",
  "#c41753",
  "#c51852",
  "#c71951",
  "#c81951",
  "#ca1a50",
  "#cb1b4f",
  "#cd1c4e",
  "#ce1d4e",
  "#cf1e4d",
  "#d11f4c",
  "#d2204c",
  "#d3214b",
  "#d5224a",
  "#d62449",
  "#d72549",
  "#d82748",
  "#d92847",
  "#db2946",
  "#dc2b46",
  "#dd2c45",
  "#de2e44",
  "#df2f44",
  "#e03143",
  "#e13342",
  "#e23442",
  "#e33641",
  "#e43841",
  "#e53940",
  "#e63b40",
  "#e73d3f",
  "#e83f3f",
  "#e8403e",
  "#e9423e",
  "#ea443e",
  "#eb463e",
  "#eb483e",
  "#ec4a3e",
  "#ec4c3e",
  "#ed4e3e",
  "#ed503e",
  "#ee523f",
  "#ee543f",
  "#ef5640",
  "#ef5840",
  "#ef5a41",
  "#f05c42",
  "#f05e42",
  "#f06043",
  "#f16244",
  "#f16445",
  "#f16646",
  "#f26747",
  "#f26948",
  "#f26b49",
  "#f26d4b",
  "#f26f4c",
  "#f3714d",
  "#f3734e",
  "#f37450",
  "#f37651",
  "#f37852",
  "#f47a54",
  "#f47c55",
  "#f47d57",
  "#f47f58",
  "#f4815a",
  "#f4835b",
  "#f4845d",
  "#f4865e",
  "#f58860",
  "#f58a61",
  "#f58b63",
  "#f58d64",
  "#f58f66",
  "#f59067",
  "#f59269",
  "#f5946b",
  "#f5966c",
  "#f5976e",
  "#f59970",
  "#f69b71",
  "#f69c73",
  "#f69e75",
  "#f6a077",
  "#f6a178",
  "#f6a37a",
  "#f6a47c",
  "#f6a67e",
  "#f6a880",
  "#f6a981",
  "#f6ab83",
  "#f6ad85",
  "#f6ae87",
  "#f6b089",
  "#f6b18b",
  "#f6b38d",
  "#f6b48f",
  "#f6b691",
  "#f6b893",
  "#f6b995",
  "#f6bb97",
  "#f6bc99",
  "#f6be9b",
  "#f6bf9d",
  "#f6c19f",
  "#f7c2a2",
  "#f7c4a4",
  "#f7c6a6",
  "#f7c7a8",
  "#f7c9aa",
  "#f7caac",
  "#f7ccaf",
  "#f7cdb1",
  "#f7cfb3",
  "#f7d0b5",
  "#f8d1b8",
  "#f8d3ba",
  "#f8d4bc",
  "#f8d6be",
  "#f8d7c0",
  "#f8d9c3",
  "#f8dac5",
  "#f8dcc7",
  "#f9ddc9",
  "#f9dfcb",
  "#f9e0cd",
  "#f9e2d0",
  "#f9e3d2",
  "#f9e5d4",
  "#fae6d6",
  "#fae8d8",
  "#fae9da",
  "#faebdd"
];
var MAKO = [
  "#0b0405",
  "#0d0406",
  "#0e0508",
  "#0f0609",
  "#10060a",
  "#11070c",
  "#12080d",
  "#13090f",
  "#140910",
  "#150a12",
  "#160b13",
  "#170c15",
  "#180d16",
  "#190e18",
  "#1a0e19",
  "#1b0f1a",
  "#1c101c",
  "#1d111d",
  "#1e111f",
  "#1f1220",
  "#201322",
  "#211423",
  "#221425",
  "#231526",
  "#241628",
  "#251729",
  "#26172b",
  "#27182d",
  "#28192e",
  "#291930",
  "#291a31",
  "#2a1b33",
  "#2b1c35",
  "#2c1c36",
  "#2d1d38",
  "#2e1e39",
  "#2e1e3b",
  "#2f1f3d",
  "#30203e",
  "#312140",
  "#312142",
  "#322243",
  "#332345",
  "#342447",
  "#342548",
  "#35254a",
  "#35264c",
  "#36274d",
  "#37284f",
  "#372851",
  "#382953",
  "#382a54",
  "#392b56",
  "#3a2c58",
  "#3a2c59",
  "#3b2d5b",
  "#3b2e5d",
  "#3b2f5f",
  "#3c3060",
  "#3c3162",
  "#3d3164",
  "#3d3266",
  "#3e3367",
  "#3e3469",
  "#3e356b",
  "#3f366d",
  "#3f366f",
  "#3f3770",
  "#403872",
  "#403974",
  "#403a76",
  "#403b78",
  "#403c79",
  "#413d7b",
  "#413e7d",
  "#413e7f",
  "#413f80",
  "#414082",
  "#414184",
  "#414285",
  "#414387",
  "#414488",
  "#40468a",
  "#40478b",
  "#40488d",
  "#40498e",
  "#3f4a8f",
  "#3f4b90",
  "#3f4c92",
  "#3e4d93",
  "#3e4f94",
  "#3e5095",
  "#3d5195",
  "#3d5296",
  "#3c5397",
  "#3c5598",
  "#3b5698",
  "#3b5799",
  "#3b589a",
  "#3a599a",
  "#3a5b9b",
  "#3a5c9b",
  "#395d9c",
  "#395e9c",
  "#385f9c",
  "#38619d",
  "#38629d",
  "#38639d",
  "#37649e",
  "#37659e",
  "#37669e",
  "#37689f",
  "#36699f",
  "#366a9f",
  "#366b9f",
  "#366ca0",
  "#366da0",
  "#366fa0",
  "#3670a0",
  "#3671a0",
  "#3572a1",
  "#3573a1",
  "#3574a1",
  "#3575a1",
  "#3576a2",
  "#3578a2",
  "#3579a2",
  "#357aa2",
  "#357ba3",
  "#357ca3",
  "#357da3",
  "#357ea4",
  "#347fa4",
  "#3480a4",
  "#3482a4",
  "#3483a5",
  "#3484a5",
  "#3485a5",
  "#3486a5",
  "#3487a6",
  "#3488a6",
  "#3489a6",
  "#348ba6",
  "#348ca7",
  "#348da7",
  "#348ea7",
  "#348fa7",
  "#3490a8",
  "#3491a8",
  "#3492a8",
  "#3493a8",
  "#3495a9",
  "#3496a9",
  "#3497a9",
  "#3498a9",
  "#3499aa",
  "#349aaa",
  "#359baa",
  "#359caa",
  "#359eaa",
  "#359fab",
  "#35a0ab",
  "#35a1ab",
  "#36a2ab",
  "#36a3ab",
  "#36a4ab",
  "#37a5ac",
  "#37a6ac",
  "#37a8ac",
  "#38a9ac",
  "#38aaac",
  "#39abac",
  "#39acac",
  "#3aadac",
  "#3aaead",
  "#3bafad",
  "#3cb1ad",
  "#3cb2ad",
  "#3db3ad",
  "#3eb4ad",
  "#3fb5ad",
  "#3fb6ad",
  "#40b7ad",
  "#41b8ad",
  "#42b9ad",
  "#43baad",
  "#44bcad",
  "#45bdad",
  "#46bead",
  "#47bfad",
  "#48c0ad",
  "#49c1ad",
  "#4bc2ad",
  "#4cc3ad",
  "#4dc4ad",
  "#4fc5ad",
  "#50c6ad",
  "#52c7ad",
  "#53c9ad",
  "#55caad",
  "#57cbad",
  "#59ccad",
  "#5bcdad",
  "#5ecdad",
  "#60ceac",
  "#62cfac",
  "#65d0ad",
  "#68d1ad",
  "#6ad2ad",
  "#6dd3ad",
  "#70d4ad",
  "#73d4ad",
  "#76d5ae",
  "#79d6ae",
  "#7cd6af",
  "#7fd7af",
  "#82d8b0",
  "#85d9b1",
  "#88d9b1",
  "#8bdab2",
  "#8edbb3",
  "#91dbb4",
  "#94dcb5",
  "#96ddb5",
  "#99ddb6",
  "#9cdeb7",
  "#9edfb8",
  "#a1dfb9",
  "#a4e0bb",
  "#a6e1bc",
  "#a9e1bd",
  "#abe2be",
  "#aee3c0",
  "#b0e4c1",
  "#b2e4c2",
  "#b5e5c4",
  "#b7e6c5",
  "#b9e6c7",
  "#bbe7c8",
  "#bee8ca",
  "#c0e9cc",
  "#c2e9cd",
  "#c4eacf",
  "#c6ebd1",
  "#c8ecd2",
  "#caedd4",
  "#ccedd6",
  "#ceeed7",
  "#d0efd9",
  "#d2f0db",
  "#d4f1dc",
  "#d6f1de",
  "#d8f2e0",
  "#daf3e1",
  "#dcf4e3",
  "#def5e5"
];
var GREY = [
  "#000000",
  "#111111",
  "#222222",
  "#333333",
  "#444444",
  "#555555",
  "#666666",
  "#777777",
  "#888888",
  "#999999",
  "#aaaaaa",
  "#bbbbbb",
  "#cccccc",
  "#dddddd",
  "#eeeeee",
  "#ffffff"
];
var CMOCEAN_PHASE = [
  "#a8780d",
  "#b3701b",
  "#be6828",
  "#c75f35",
  "#cf5643",
  "#d54b53",
  "#db4066",
  "#de357b",
  "#df2a93",
  "#dc25ad",
  "#d529c4",
  "#cc34d7",
  "#c041e5",
  "#b24fef",
  "#a25cf3",
  "#9168f4",
  "#7d73f0",
  "#687ce8",
  "#5285dc",
  "#3d8bcd",
  "#2c90bc",
  "#2093ac",
  "#19959c",
  "#12978c",
  "#0c987c",
  "#119a69",
  "#249a52",
  "#409839",
  "#5e9420",
  "#778d12",
  "#8b860d",
  "#9b7f0d"
];
var reversePalette2 = (palette) => palette.slice().reverse();
var SEQUENTIAL_PALETTES = {
  turbo: TURBO,
  // rainbow-like
  "turbo-reversed": reversePalette2(TURBO),
  grey: GREY,
  "grey-reversed": reversePalette2(GREY),
  magma: MAGMA,
  "magma-reversed": reversePalette2(MAGMA),
  inferno: INFERNO,
  "inferno-reversed": reversePalette2(INFERNO),
  plasma: PLASMA,
  // yellow to purple
  "plasma-reversed": reversePalette2(PLASMA),
  viridis: VIRIDIS,
  "viridis-reversed": reversePalette2(VIRIDIS),
  cividis: CIVIDIS,
  "cividis-reversed": reversePalette2(CIVIDIS),
  rocket: ROCKET,
  //
  "rocket-reversed": reversePalette2(ROCKET),
  mako: MAKO,
  // blue to teal
  "mako-reversed": reversePalette2(MAKO),
  "cmocean-phase": CMOCEAN_PHASE,
  "cmocean-phase-reversed": reversePalette2(CMOCEAN_PHASE)
};

// src/palettes/swatches.ts
var BACKGROUND_COLORS = {
  red: "#fbbbb6",
  orange: "#ffe0c2",
  yellow: "#fff1c2",
  lime: "#d0e8b9",
  green: "#bceac4",
  teal: "#b9f1f1",
  cyan: "#b8e5c9",
  blue: "#b6d9fb",
  indigo: "#d1c2f0",
  purple: "#e3baf8",
  magenta: "#f9c8e0",
  black: "#353535",
  "dark-grey": "#8C8C8C",
  grey: "#D0D0D0",
  "light-grey": "#F0F0F0",
  white: "#ffffff"
};
var FOREGROUND_COLORS = {
  red: "#d7170b",
  //<- 700, 500 ->'#f21c0d'
  orange: "#fe8a2b",
  yellow: "#ffc02b",
  // <- 600, 500 -> '#ffcf33',
  lime: "#63b215",
  green: "#21ba3a",
  teal: "#17cfcf",
  cyan: "#13a7ec",
  blue: "#0d80f2",
  indigo: "#63c",
  purple: "#a219e6",
  magenta: "#eb4799",
  black: "#000",
  "dark-grey": "#666",
  grey: "#A6A6A6",
  "light-grey": "#d4d5d2",
  white: "#ffffff"
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  BACKGROUND_COLORS,
  CATEGORICAL_PALETTES,
  COLOR_SCALE_PRESETS,
  DIVERGING_PALETTES,
  FOREGROUND_COLORS,
  SEQUENTIAL_PALETTES,
  apca,
  asColorNumber,
  asRgb,
  blue,
  brown,
  cMaxFor,
  colorToCss,
  colorToHex,
  contrastingColor,
  cyan,
  darkShades,
  darken,
  darkenHex,
  gammaCorrect,
  getColorScaleMode,
  getLightShadeMap,
  getOptimalHueRange,
  gray,
  green,
  hslToRgb,
  indigo,
  interpolateOklch,
  inverseGammaCorrect,
  isLightColor,
  isNamedColor,
  isValidColor,
  lerpOklch,
  liftColor,
  lighten,
  lime,
  mixColors,
  oklabToRgb,
  oklch,
  oklchFromRGB,
  oklchToRgb,
  orange,
  parseColor,
  parseColorToHex,
  parseColorToRgb,
  parseColorToRgb01,
  parseColorToRgba01,
  parseColorWithAlpha,
  parseHexColor,
  pink,
  purple,
  red,
  rgbToHex,
  rgbToHsl,
  rgbToOklab,
  rgbToOklch,
  shade,
  shades,
  teal,
  toDarkColor,
  yellow
});
